core/analyze.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright © 2022 The K9CLI Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	v4 "github.com/aws/aws-sdk-go-v2/aws/signer/v4"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"time"
+)
+
+type analyzeRequestBody struct {
+	CustomerID string `json:"customerId"`
+	Account    string `json:"accountId"`
+}
+
+// Example:
+// {"customerId": "C10000", "accountId": "139710491120", "executionId": "ondemand-C10000-139710491120-2022-09-15_TV49"}
+type analyzeResponseBody struct {
+	CustomerID string `json:"customerId"`
+	Account    string `json:"accountId"`
+	ExecutionID string `json:"executionId"`
+}
+
+func AnalyzeAccount(o io.Writer, cfg aws.Config, apiHost, customerID, account string) error {
+	url := fmt.Sprintf("https://%s/analysis/account", apiHost)
+
+	requestBody := analyzeRequestBody{
+		CustomerID: customerID,
+		Account:    account,
+	}
+	requestBodyBytes, _ := json.Marshal(requestBody)
+
+	request, err := http.NewRequest("POST", url, bytes.NewBuffer(requestBodyBytes))
+	now := time.Now()
+	request.Header.Set("Date", now.Format(time.RFC3339))
+	request.Header.Set("Content-Type", "application/json")
+
+	err = signApiRequest(cfg, request, now)
+	if err != nil {
```

Comment:
Gotta handle the errors.

Reply:
And consider condensing this to a 1-liner because you're not using multiple assignment:

```go
	if err = signApiRequest(cfg, request, now); err != nil {
		// do something
	}```

Reply:
consolidated to 1-line form